test/problems/poisson_regression.jl

```diff
@@ -0,0 +1,239 @@
+# OptimizationMethods.jl 
+
+module TestPoissonRegression 
+
+using Test, ForwardDiff, OptimizationMethods, Random, NLPModels
+
+@testset "Problem: Poission Regression" begin
+
+    # set the seed for reproducibility
+    Random.seed!(1010)
+
+    ####################################
+    # Test Struct: Poisson Regression
+    ####################################
+
+    # Check if struct is defined 
+    @test isdefined(OptimizationMethods, :PoissonRegression)
+
+    # Test Super type 
+    @test supertype(OptimizationMethods.PoissonRegression) == AbstractNLPModel
+
+    # Test Fields 
+    for name in [:meta, :counters, :design, :response]
+        @test name in fieldnames(OptimizationMethods.PoissonRegression)
+    end
+
+    
+    # Test Constructor
+    real_types = [Float16, Float32, Float64]
+    nobs_default = 1000
+    nvar_default = 50
+
+    for real_type in real_types
+        #Check Assertions 
+        @test_throws AssertionError OptimizationMethods.PoissonRegression(
+            real_type, nobs=-1, nvar=50)
+        @test_throws AssertionError OptimizationMethods.PoissonRegression(
+            real_type, nobs=1000, nvar=-1)
+
+        #Generate Random Problem
+        progData = OptimizationMethods.PoissonRegression(real_type)
+        
+        #Check design type and dimensions
+        @test typeof(progData.design) == Matrix{real_type}
+        @test size(progData.design) == (nobs_default, nvar_default)
+        
+        #Check response type and dimensions 
+        @test typeof(progData.response) == Vector{real_type}
+        @test length(progData.response) == nobs_default
+        
+        #Check default initial guess and dimensions
+        @test typeof(progData.meta.x0) == Vector{real_type}
+        @test length(progData.meta.x0) == nvar_default
+    end
+
```

Comment:
Tests for the second constructor required.

Reply:
Done. Did not include checking of values and types as these are supplied.